src/kem/ml_kem/icicle_ml-kem-1024_icicle_cuda/icicle_ml-kem.cpp

```diff
@@ -0,0 +1,41 @@
+#include <stdint.h>
+#include <oqs/rand.h>
+#include "icicle/runtime.h"
+#include <icicle/pqc/ml_kem.h>
+
+extern "C" {
+int icicle_ml_kem_1024_keypair(uint8_t *pk, uint8_t *sk) {
+    uint8_t coins[icicle::pqc::ml_kem::ENTROPY_BYTES];
+    OQS_randombytes(coins, icicle::pqc::ml_kem::ENTROPY_BYTES);
+#if defined(OQS_ENABLE_KEM_ml_kem_1024_icicle_cuda)
+    icicle::Device dev = {"CUDA-PQC", 0};
+#endif
+    icicle_set_device(dev);
+
+    icicle::pqc::ml_kem::MlKemConfig config;
+
+    return (int)icicle::pqc::ml_kem::keygen<icicle::pqc::ml_kem::Kyber1024Params>((std::byte*)coins, config, (std::byte*)pk, (std::byte*)sk);
```

Comment:
here you case eIcicleResult to int? maybe compare to SUCCESS and return 0/1 accordingly

Reply:
@yshekel Is that what you mean?

```suggestion
    return (int)(icicle::pqc::ml_kem::keygen<icicle::pqc::ml_kem::Kyber1024Params>((std::byte*)coins, config, (std::byte*)pk, (std::byte*)sk) != eIcicleError::SUCCESS);
```